h5py/tests/test_dtype.py

```diff
@@ -289,13 +291,14 @@ def test_float_round_tripping(self):
                      if (np.issubdtype(f, np.floating) or
                          np.issubdtype(f, np.complexfloating)))
 
+        unsupported_types = []
         if platform.machine() in UNSUPPORTED_LONG_DOUBLE:
-            dtype_dset_map = {str(j): d
-                              for j, d in enumerate(dtypes)
-                              if d not in (np.float128, np.complex256)}
-        else:
-            dtype_dset_map = {str(j): d
-                              for j, d in enumerate(dtypes)}
+            for x in UNSUPPORTED_LONG_DOUBLE_TYPES:
+                if hasattr(np, x):
+                    unsupported_types.append(getattr(np, x))
+        dtype_dset_map = {str(j): d
+                          for j, d in enumerate(dtypes)
+                          if d not in unsupported_types}
```

Comment:
I'm not sure why we need the additional list `unsupported_types`, I'd expect `if d not in UNSUPPORTED_LONG_DOUBLE_TYPES` should be sufficient (we could also make them sets rather than tuples, but not sure that would actually cause a speed up given the small number of iterms)?

Reply:
`UNSUPPORTED_LONG_DOUBLE_TYPES` are strings while `d` is the actual type.

Reply:
Ah, ok.